### PR TITLE
server: fix issue 2346

### PR DESCRIPTION
**Pull request by tiancaiamao**
Fix https://github.com/pingcap/tidb/issues/2346

ComStmtExecute packet just send params type information the first time,
so PrepareStmt should save it for further use.

I've test is manually, but it's hard to add test case because Go driver just set the param flag 
and send the parameters type every time.

### CODE REVIEW

File: server/conn_stmt.go [hunk 1]
```diff
@@ -151,9 +151,14 @@ func (cc *clientConn) handleStmtExecute(data []byte) (err error) {
 			paramTypes = data[pos : pos+(numParams<<1)]
 			pos += (numParams << 1)
 			paramValues = data[pos:]
+			// Just the first StmtExecute packet contain parameters type,
+			// we need save it for further use.
+			stmt.SetParamsType(paramTypes)
```
Comment (coocood) [inline server/conn_stmt.go]: The parameters can be sent separately.
Say there are two params 'a' and 'b', 'a' is sent first, then 'b' is sent, if we set paramTypes like this, the type of  param 'a' would be lost.

Comment (tiancaiamao) [inline server/conn_stmt.go]: Your example is a invalid case, client should not sent 'a' first, then 'b' ...

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**